codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -579,7 +428,8 @@ private TextHeaders addEscapedValue(CharSequence name, CharSequence escapedValue
         }
 
         private <T> CharSequence commaSeparate(CsvValueEscaper<T> escaper, T... values) {
-            StringBuilder sb = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
```

Comment:
I don't see how we are decreasing complexity here.  If anything seems like we are just masking complexity.  If it is upfront perhaps someone may be able to make it easier to notice, or potentially easier to configure.  I think asking questions is a benefit of making it visible...there is no "right" choice for this and it is likely to be "better" choices for different use cases.  Perhaps moving it into a utility class may be more appropriate?